pinot/regressors/biophysical_regressor.py

```diff
@@ -23,95 +24,103 @@ class BiophysicalRegressor(torch.nn.Module):
-        self.base_regressor = base_regressor
+        # get the base regressor
+        self.base_regressor_class = base_regressor_class
+        self.base_regressor = base_regressor_class(
+                *args, **kwargs
+        )
+
+        # initialize measurement parameter
         self.log_sigma_measurement = torch.nn.Parameter(torch.zeros(1))
 
-    def g(self, func_value=None, test_ligand_concentration=1e-3):
-        return 1 / (1 + torch.exp(-func_value) / test_ligand_concentration)
+    def _get_measurement(self, delta_g, concentration=1e-3):
+        """ Translate ..math:: \Delta G to percentage inhibition.
+
+        Parameters
+        ----------
+        delta_g : torch.Tensor, shape=(number_of_graphs, 1)
+           Binding free energy. 
+
+        concentration : torch.Tensor, shape=(,) or (1, number_of_concentrations)
+            Concentration of ligand.
+
+        Returns
+        -------
+        measurement : torch.Tensor, shape=(number_of_graphs, 1)
+            or (number_of_graphs, number_of_concentrations)
+            Percentage of inhibition.
+
+        """
+        return 1.0 / (1.0 + torch.exp(-delta_g) / concentration)
+
+    def _condition_delta_g(self, x_te, *args, **kwargs):
+        """ Returns predictive distribution of binding free energy. """
+        return self.base_regressor.condition(x_te, *args, **kwargs)
+
+    def _condition_measurement(self, x_te=None, concentration=1e-3, delta_g_sample=None):
+        """ Returns predictive distribution of percentage of inhibtiion. """
+        # sample delta g if not specified
+        if delta_g_sample is None and x_te is not None:
+            delta_g_sample = self._condition_delta_g(x_te).rsample()
 
-    def condition(
-        self, h=None, test_ligand_concentration=1e-3, *args, **kwargs
-    ):
-        distribution_base_regressor = self.base_regressor.condition(
-            h, *args, **kwargs
-        )
-        # we sample from the latent f to push things through the likelihood
-        # Note: if we do this,
-        # in order to get good estimates of LLK
-        # we may need to draw multiple samples
-        f_sample = distribution_base_regressor.rsample()
-        mu_m = self.g(
-            func_value=f_sample,
-            test_ligand_concentration=test_ligand_concentration,
-        )
-        sigma_m = torch.exp(self.log_sigma_measurement)
         distribution_measurement = torch.distributions.normal.Normal(
-            loc=mu_m, scale=sigma_m
+            loc=self._get_measurement(delta_g_sample, concentration=concentration),
+            scale=self.log_sigma_measurement.exp()
         )
-        # import pdb; pdb.set_trace()
-        return distribution_measurement
 
-    def loss(
-        self, h=None, y=None, test_ligand_concentration=None, *args, **kwargs
+    def _condition_ic_50(
+            self, 
+            x_te=None,
+            delta_g_sample=None,
+            concentration_low=0.0,
+            concentration_high=1.0,
```

Comment:
What do these parameters mean? Can you document them in a docstring?

We _definitely_ do not want a uniform distribution over IC50 as a prior. It should be uniform over `log10(IC50)`, over a meaningful physical range. You can see what a sensible range is from this paper:
https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061007

![image](https://user-images.githubusercontent.com/3656088/87868010-07445080-c947-11ea-9c0d-401d668c3d4e.png)

So maybe 0-14 in `-log10(IC50)`.

Reply:
Thanks John, I think this was sth @yuanqing-wang just wanted to ask me stuff about that he hacked in just today. 
@yuanqing-wang can you elaborate on your thinking? Ideally also with PR comments when you add a new function like this?

Reply:
Sorry @jchodera this is just a very immature draft I had in order to discuss some ideas. For the IC50 I was only thinking about how can we integrate that into the graphical model without using the Cheng-Prussoff equation. So I'm thinking about doing the titration curve here from `concentration_low` to `concentration_high`. But I realized that I can only do sampling this way but not get the distribution.